otherlibs/win32unix/select.c

```diff
@@ -960,19 +960,19 @@ static int fdlist_to_fdset(value fdlist, fd_set *fdset)
 
 static value fdset_to_fdlist(value fdlist, fd_set *fdset)
 {
-  value res = Val_int(0);
-  Begin_roots2(fdlist, res)
-    for (/*nothing*/; fdlist != Val_int(0); fdlist = Field(fdlist, 1)) {
-      value s = Field(fdlist, 0);
-      if (FD_ISSET(Socket_val(s), fdset)) {
-        value newres = caml_alloc_small(2, 0);
-        Field(newres, 0) = s;
-        Field(newres, 1) = res;
-        res = newres;
-      }
+  CAMLparam1(fdlist);
+  CAMLlocal2(res, s);
+  res = Val_int(0);
+  for (/*nothing*/; fdlist != Val_int(0); fdlist = Field(fdlist, 1)) {
+    s = Field(fdlist, 0);
+    if (FD_ISSET(Socket_val(s), fdset)) {
+      value newres = caml_alloc_small(2, 0);
+      Field(newres, 0) = s;
+      Field(newres, 1) = res;
+      res = newres;
```

Comment:
For the record: the original code (with `Begin_roots2`) was modeled after the `fdset_to_fdlist` function from `otherlibs/unix/select.c`, which still uses... old-style `Begin_roots2`/`End_roots` root management.

Reply:
Indeed - I think it's OK only to "upgrade" these macros when something's actually changed/corrected?

Reply:
(the Unix version obviously has no problem, since the representation of fd's is simpler)